[Winston Leonard Prayonggo] iP

WinstonLeonard opened the pull request:
# Duke Chatbot
> "Your mind is for having ideas, not holding them." -David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke Chatbot frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
*  ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/WinstonLeonard/ip). 
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

#

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

WinstonLeonard committed:
Level-0

WinstonLeonard committed:
Update Duke.java

WinstonLeonard committed:
Level-1

WinstonLeonard committed:
Level-2

WinstonLeonard committed:
Mark as Done

WinstonLeonard committed:
ToDo, Event, Deadline

WinstonLeonard committed:
Automated Text UI Testing

WinstonLeonard committed:
Handle Errors

WinstonLeonard committed:
Delete

WinstonLeonard committed:
before adding handlers

WinstonLeonard committed:
Save

WinstonLeonard committed:
Merge pull request #1 from WinstonLeonard/branch-Level-7

Branch level 7

WinstonLeonard committed:
fixed convertDeadlineFormatter

WinstonLeonard committed:
fixed mark and unmark

WinstonLeonard committed:
Merge pull request #2 from WinstonLeonard/branch-Level-8

Branch level 8

WinstonLeonard committed:
MoreOOP completed

deleted comments

WinstonLeonard committed:
deleted more comments

WinstonLeonard committed:
Merge pull request #3 from WinstonLeonard/MoreOOP

More oop

WinstonLeonard committed:
Moved all classes into Duke Package

WinstonLeonard committed:
Merge pull request #4 from WinstonLeonard/Packages

Moved all classes into Duke Package

WinstonLeonard committed:
Merge pull request #5 from WinstonLeonard/add-gradle-support

Add gradle support

WinstonLeonard committed:
Run with Gradle

WinstonLeonard committed:
Merge pull request #6 from WinstonLeonard/Gradle

Gradle

WinstonLeonard committed:
Added testcases for Deadline and ToDo

WinstonLeonard committed:
Merge pull request #7 from WinstonLeonard/JUnit

Added testcases for Deadline and ToDo

WinstonLeonard committed:
created jar

WinstonLeonard committed:
Merge pull request #8 from WinstonLeonard/Jar

created jar

WinstonLeonard committed:
added Find

WinstonLeonard committed:
Merge pull request #9 from WinstonLeonard/Find

added Find

WinstonLeonard committed:
added java docs to all public methods

WinstonLeonard committed:
Merge pull request #10 from WinstonLeonard/java-docs-and-styling

added java docs to all public methods

WinstonLeonard committed:
JavaFX tutorial part 3

bobscodedump commented on src/main/java/duke/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,8 @@
+public class EmptyDescriptionException extends DukeException {
+    public static String line = "\t____________________________________________________________\n";
+    public EmptyDescriptionException(String task) {
+        //System.out.println( line + "\t☹ OOPS!!! The description of a " + task + " cannot be empty." + "\n" + line);
```
You could remove commented out code before submitting PR

bobscodedump commented on src/main/java/duke/NotANumberException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+public class NotANumberException extends DukeException {
+    public static String line = "\t____________________________________________________________\n";
+
```
inconsistent spacing with other Exception classes

bobscodedump commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+    /**
+     * Method to write to the storage
+     */
+
```
inconsistent spacing with javadocs

bobscodedump commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+public class TaskList {
+    public static String line = "\t____________________________________________________________\n";
+    protected ArrayList<Task> strList;
+
```
inconsistent spacing between methods

bobscodedump commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+    /**
+     * Constructor to create a new task list
+     */
+    public TaskList() {
```
You could put your constructor between your variable declarations and methods

bobscodedump reviewed (commented):
Great submission overall, there are just some minor inconsistencies with spacing and other small things like javadocs.

WinstonLeonard committed:
added padding, cropped image into circle, add background color

marcellaantania commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,22 @@
+        ui.startUi();
+        storage.writeToStorage();
+    }
+    public static void main(String[] args) throws EmptyDescriptionException, InvalidCommandException, NotANumberException {
```
It would be better to line break this to improve readability as this has 123 characters including indentation.

WinstonLeonard committed:
added MainWindow.fxml

marcellaantania commented on src/main/java/duke/NotANumberException.java:
```diff
@@ -0,0 +1,8 @@
+package duke;
+public class NotANumberException extends DukeException {
```
I like this additional exception you made! This slipped my mind when I built my own one.

marcellaantania commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+                String isDone = Character.toString(fileLine.charAt(4));
+                String extractedSubstring = fileLine.substring(7, fileLine.length());
+                switch (eventType) {
+                    case "T":
```
I believe "case" should not be indented relative to "switch" in this module

WinstonLeonard committed:
Completed GUI

marcellaantania commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+     * @param readingFile boolean to check if readingFile or not
+     * @throws EmptyDescriptionException
+     */
+    public void toDoHandler(String description, boolean isDone, boolean readingFile) throws EmptyDescriptionException {
```
I think the method name can be more intuitive as I was not sure what kind of handling this method does. I thought it was only a checker of the validity of todo's parameters until I realized that it adds the todo to the TaskList at the end.

marcellaantania commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+     * Prints out all the tasks that match the taskName
+     * @param taskName the name of the task that we are finding for
+     */
+    public void findTask(String taskName) {
```
As we can actually find tasks without giving the full task name, maybe the variable taskName can be changed to something like "keyword" or "keyphrase" instead? The JavaDoc can be modified as well for the same reason for better clarity.

marcellaantania reviewed (commented):
I like that your code is pretty intuitive to read and understand. There were some minor improvements that could be made, but good job overall! :)

WinstonLeonard committed:
merged GUI and UI

WinstonLeonard committed:
Merge pull request #11 from WinstonLeonard/GUI

GUI

dom-buri commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,74 @@
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+            text = l;
```
Should there be an indentation here?

dom-buri commented on src/main/java/duke/InvalidCommandException.java:
```diff
@@ -0,0 +1,8 @@
+public class InvalidCommandException extends DukeException {
+    public static String line = "\t____________________________________________________________\n";
+    public InvalidCommandException() {
+        System.out.println(line + " \t☹ OOPS!!! I'm sorry, but I don't know what that means :-(" + "\n" +line);
```
There should be a space in between + and line

dom-buri commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,70 @@
+import java.util.Scanner;
+import java.util.Arrays;
+public class Ui {
+    public static String line = "\t____________________________________________________________\n";
+    public static String exitMessage = line + "\tBye. Hope to see you again\n" + line;
```
Should the variable name be all uppercase instead as they are constants?

dom-buri commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+            e.printStackTrace();
+            System.out.println("The file named duke.txt does not exist.");
+        } catch (EmptyDescriptionException e) {
+            //do something
```
Avoid empty catch blocks. Perhaps you could add a comment to explain why the catch block is left empty.

dom-buri reviewed (commented):
LGTM. Just a few nits to fix regarding Java coding standards and code quality.

WinstonLeonard committed:
Fixed checkstyle

WinstonLeonard committed:
Merge pull request #12 from WinstonLeonard/checkstyle

Fixed checkstyle

WinstonLeonard committed:
added assertions

assertions that were added:

asserted input task number must be in the task list

asserted correct input of task type

asserted deadline task must contain /by

asserted event task must contain /to and /from

WinstonLeonard committed:
Merge pull request #13 from WinstonLeonard/branch-A-Assertions

added assertions

WinstonLeonard committed:
added circle clip class

WinstonLeonard committed:
Revert "added assertions"

This reverts commit 7a86abaefbd52e9dc933a69cb69be2ce188b3bbb.

WinstonLeonard committed:
added assertions

assertions that were added:

asserted input task number must be in the task list

asserted correct input of task type

asserted deadline task must contain /by

asserted event task must contain /to and /from

WinstonLeonard committed:
Merge branch 'branch-A-CodeQuality' of https://github.com/WinstonLeonard/ip into branch-A-CodeQuality

WinstonLeonard committed:
added CircleClip class

WinstonLeonard committed:
created EventTypeHandler class

WinstonLeonard committed:
Abstracted out Window into a new class

WinstonLeonard committed:
Merge pull request #14 from WinstonLeonard/branch-A-CodeQuality

Branch a code quality

WinstonLeonard committed:
added update for deadline and task

WinstonLeonard committed:
update time for event

Update time for event, and also made Window class methods static

WinstonLeonard committed:
Merge pull request #15 from WinstonLeonard/branch-BCD-Extension

Branch bcd extension

WinstonLeonard committed:
added Ui.png

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
added help command

WinstonLeonard committed:
Merge pull request #17 from WinstonLeonard/final-tweaks

Final tweaks

WinstonLeonard committed:
added Ui.png

WinstonLeonard committed:
Fixed styling and added more javadoc

WinstonLeonard committed:
Update README.md

WinstonLeonard committed:
Handled invalid filepath

WinstonLeonard committed:
Merge branch 'master' of https://github.com/WinstonLeonard/ip
